src/impute.jl

```diff
@@ -126,12 +126,12 @@ julia> [k => parent(parent(v)) for (k, v) in Impute.filter(ds; dims=:loc).data]
 """
 Impute.apply(ds::KeyedDataset, f::Filter; dims) = Impute.apply!(deepcopy(ds), f; dims=dims)
 
-_pattern(dims::Pattern) = dims
-_pattern(dims::Tuple) = Pattern(dims)
-_pattern(dims) = Pattern(:__, dims)
+_impute_pattern(dims::Pattern) = dims
```

Comment:
Having to distinguish `_impute_pattern` and `_transform_pattern` could be a code smell? But there is a difference in what `dims` means for Impute vs. FT.

`_transform_pattern` is closer to what `mapslices` does: https://github.com/invenia/AxisSets.jl/blob/d39b647fd0dc27954eb21f8ef047e7112910b103/src/functions.jl#L79-L84

Ideally we'd standardise how to handle patterns/dims everywhere.

Reply:
Yeah, maybe add a comment to https://github.com/invenia/Impute.jl/issues/66? If we're gonna change that in Impute.jl it'd be good to do that before a 1.0 release?

Reply:
https://github.com/invenia/Impute.jl/issues/66#issuecomment-821258045